addons/hr_expense_stripe/utils.py

```diff
@@ -470,54 +451,54 @@ def format_amount_to_stripe(amount, currency):
     """ Helper to convert currencies according from stripe formatting which is the amount in the currency's minor unit with exceptions  """
     return to_minor_currency_units(amount, currency, arbitrary_decimal_number=STRIPE_EXCEPTIONS_CURRENCY_MINOR_UNITS.get(currency.name))
 
-def stripe_make_request(api_key, endpoint, payload=None, method='POST', offline=False, idempotency_key=None):
+def flatten_dict(dict_vals, level=0):
     """
-    Make a request to Stripe API at the specified endpoint.
+    Stripe has a flat data hierarchy and do not accept objects
 
-    Note: self.ensure_one()
+    this ->     {
+                    'person': {
+                        'name': "Eli",
+                        'lastname': "Lambert", 
+                    }
+                }
 
-    :param str endpoint: The endpoint to be reached by the request
-    :param dict payload: The payload of the request
-    :param str method: The HTTP method of the request
-    :param bool offline: Whether the operation of the transaction being processed is 'offline'
-    :param str idempotency_key: The idempotency key to pass in the request.
-    :return The JSON-formatted content of the response
-    :rtype: dict
-    :raise: ValidationError if an HTTP error occurs
+    becomes ->  {
+                    'person[name]': "Eli",
+                    'person[lastname]': "Lambert"
+                }
```

Comment:
You're describing an application/x-www-form-urlencoded request, which it is
so I'd rename the method and avoid describing it too much, just say we're converting dicts into urlencoded forms